### ISSUE TITLE
[DOCS] How to upgrade a Fleet Server on premise and best practices

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/ingest-docs#30]
How to upgrade Fleet Server on premise and best practices to avoid downtime.
Might include:
- Using more than 1 Fleet Server
- How to load balance multiple Fleet Servers
- Upgrade order and if it supports rolling upgrades or not

Improved guidance for air-gapped environments.

**Comment by karenzone** [comment]
Is there a way to see who set the project fields for this (and other) issues? 

"Estimate," for example. With the limited details and scope available, no SME identified, and no input from writers, `<2 days` seems overly optimistic.

**Comment by dedemorton** [comment]
>With the limited details and scope available, no SME identified, and no input from writers, <2 days seems overly optimistic.

Agreed. AFAIK, we don't have any primary source materials to work from. If we are expecting a writer to learn how to install and upgrade Fleet Server on prem and document best practices (along with load balancing and air gapped environments), this effort is going to take more than a week.

@karenzone I've added the needs-input label though I'm not sure if your team is tracking that label these days.

**Comment by lucabelluccini** [comment]
This should also be integrated with the wider Stack docs https://www.elastic.co/guide/en/elastic-stack/7.17/upgrading-elastic-stack.html as we document upgrading Elastic Agent but not when to upgrade Fleet Server.

**Comment by nimarezainia** [comment]
Agreed this is not a docs issue, I had placed it on our backlog as we need engineering assigned to help with the details. (fyi @pierrehilbert )

@lucabelluccini for your reference new air-gap docs are now here: https://www.elastic.co/guide/en/elastic-stack/current/air-gapped-install.html

**Comment by nimarezainia** [comment]
@lucabelluccini, coming back to this. Considerable amount of effort went into defining how to install and order of upgrade in the air-gap environments: https://www.elastic.co/guide/en/fleet/8.x/air-gapped.html

In context of overall stack installation and upgrade in an air-gap/on-premise environments we also created this document: https://www.elastic.co/guide/en/elastic-stack/current/air-gapped-install.html#air-gapped-fleet   which essentially says that Fleet Server is just another agent. 


@kilfoyle perhaps we could write a paragraph to address the first few bullets in the description. I'm not sure this warrants engineering time and I can write it up. I feel like majority of the content is in the docs but perhaps not in one place. May be we can add a section here to [cover these](https://www.elastic.co/guide/en/fleet/8.x/add-fleet-server-on-prem.html) questions. wdyt?

**Comment by bmorelli25** [comment]
The majority of work here has been done. If there are improvements that are still needed, let's address them in a new issue.
